web/src/pages/documents/documents-page.tsx

```diff
@@ -150,11 +153,13 @@ const DocumentsPage = () => {
                             </div>
                         }
                         sidebarHeaderContent={
-                            <DocumentsPageControlConnector
-                                isSearchPage
-                                handleUploadWizardButtonClick={handleUploadWizardButtonClick}
-                                onSearchClick={onSearchClick}
-                            />
+                            <UploadIndicatorContextProvider>
```

Comment:
I think we can achieve our goal without creating of context. Upload button is just used in 1 place and this logic can be used there. In case if we apply the approach I proposed with moving the logic to the separate `fetchMethod` callback, we can achieve this.

Reply:
Context removed

Reply:
Checked